internal/docs/sample_event.go

```diff
@@ -54,7 +55,7 @@ func renderSampleEvent(packageRoot, dataStreamName string) (string, error) {
 			stripDataStreamFolderSuffix(dataStreamName)))
 	}
 	builder.WriteString("```json\n")
-	builder.Write(formatted)
+	builder.Write(bytes.TrimSpace(formatted))
```

Comment:
To avoid having different results in existing packages, wdyt about making this dependant on the spec version? This way the change in packages is controlled by a change in the spec. This is already done for other cases now.
```suggestion
	if !specVersion.LessThan(semver.MustParse("3.2.3")) {
		builder.Write(bytes.TrimSpace(formatted))
	}
```

Reply:
This is not necessary.

Reply:
How can we make this change compatible with existing packages then?

Even if we merge https://github.com/elastic/integrations/pull/11174, older packages, and older versions of elastic-package will be incompatible.

Reply:
It is already compatible, _if the package conforms to the ep generated format_. This is the point of the preparatory change. Once we get everything into the canonical form, the two states of this code (this change and the previous version) both produce the same README files and so the checks will pass. The sample files will differ, but they are not checked, which is essentially why the preparatory change is required.